apps/pie-storybook/stories/testing/pie-modal.test.stories.ts

```diff
@@ -0,0 +1,336 @@
+import { html } from 'lit/static-html.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { type Meta } from '@storybook/web-components';
+
+import '@justeattakeaway/pie-modal';
+import {
+  type PieModal, type ModalProps as ModalPropsBase, headingLevels, sizes, positions, defaultProps,
+} from '@justeattakeaway/pie-modal';
+
+import '@justeattakeaway/pie-button';
+import '@justeattakeaway/pie-text-input';
+import '@justeattakeaway/pie-form-label';
+
+import { type SlottedComponentProps } from '../../types';
+import { createStory, sanitizeAndRenderHTML } from '../../utilities';
+
+type ModalProps = SlottedComponentProps<ModalPropsBase>;
+type ModalStoryMeta = Meta<ModalProps>;
+
+const defaultArgs: ModalProps = {
+  ...defaultProps,
+  hasBackButton: false,
+  heading: 'This is a modal heading',
+  isDismissible: false,
+  isFooterPinned: true,
+  isOpen: true,
+  slot: '',
+  leadingAction: {
+    text: '',
+    variant: undefined,
+    ariaLabel: '',
+  },
+  supportingAction: {
+    text: '',
+    variant: undefined,
+    ariaLabel: '',
+  },
+  aria: {
+    back: 'Back',
+    close: 'Close',
+    loading: 'Loading',
+  },
```

Comment:
Some of these values are already set in `defaultProps`
```suggestion
  heading: 'This is a modal heading',
  isOpen: true,
  slot: '',
  leadingAction: {
    text: '',
    variant: undefined,
    ariaLabel: '',
  },
  supportingAction: {
    text: '',
    variant: undefined,
    ariaLabel: '',
  },
  aria: {
    back: 'Back',
    close: 'Close',
    loading: 'Loading',
  },
```